it/src/test/java/com/linecorp/centraldogma/it/mirror/git/LocalToRemoteGitMirrorTest.java

```diff
@@ -0,0 +1,523 @@
+                .hasMessageContaining("contains more than")
+                .hasMessageContaining("file");
+    }
+
+    @Test
+    void localToRemote_tooManyBytes() throws Exception {
+        pushMirrorSettings(null, null, null);
+
+        // Add files whose total size exceeds the allowed maximum.
+        long remainder = MAX_NUM_BYTES + 1;
+        final int defaultFileSize = (int) (MAX_NUM_BYTES / MAX_NUM_FILES * 2);
+        final ArrayList<Change<String>> changes = new ArrayList<>();
+        for (int i = 0;; i++) {
+            final int fileSize;
+            if (remainder > defaultFileSize) {
+                remainder -= defaultFileSize;
+                fileSize = defaultFileSize;
+            } else {
+                fileSize = (int) remainder;
+                remainder = 0;
+            }
+
+            changes.add(Change.ofTextUpsert("/" + i + ".txt", Strings.repeat("*", fileSize)));
+
+            if (remainder == 0) {
+                break;
+            }
+        }
+        client.forRepo(projName, REPO_FOO).commit("Add a bunch of numbered asterisks", changes).push().join();
+
+        // Perform mirroring, which should fail.
+        assertThatThrownBy(() -> mirroringService.mirror().join())
+                .hasCauseInstanceOf(MirrorException.class)
+                .hasMessageContaining("contains more than")
+                .hasMessageContaining("byte");
+    }
+
+    @CsvSource({ "meta", "dogma" })
+    @ParameterizedTest
+    void cannotMirrorInternalRepositories(String localRepo) {
+        assertThatThrownBy(() -> pushMirrorSettings(localRepo, "/", "/", null, MirrorDirection.LOCAL_TO_REMOTE))
+                .hasCauseInstanceOf(CentralDogmaException.class)
+                .hasMessageContaining("invalid localRepo:");
+    }
+
+    private void pushMirrorSettings(@Nullable String localPath, @Nullable String remotePath,
+                                    @Nullable String gitignore) {
+        pushMirrorSettings(REPO_FOO, localPath, remotePath, gitignore, MirrorDirection.LOCAL_TO_REMOTE);
+    }
+
+    private void pushMirrorSettings(String localRepo, @Nullable String localPath, @Nullable String remotePath,
+                                    @Nullable String gitignore, MirrorDirection direction) {
+        client.forRepo(projName, Project.REPO_META)
+              .commit("Add /mirrors.json",
+                      Change.ofJsonUpsert("/mirrors.json",
+                                          "[{" +
+                                          "  \"type\": \"single\"," +
+                                          "  \"direction\": \"" + direction + "\"," +
+                                          "  \"localRepo\": \"" + localRepo + "\"," +
+                                          (localPath != null ? "\"localPath\": \"" + localPath + "\"," : "") +
+                                          "  \"remoteUri\": \"" + gitUri + firstNonNull(remotePath, "") + '"' +
```

Comment:
What do you think of also adding this to `GitMirrorTest` as well?

```suggestion
                                          "  \"remoteUri\": \"" + gitUri + firstNonNull(remotePath, "") + '"' +
                                          "  \"schedule\": \"0 0 0 1 1 ? 2099\"," +
```